src/printer.js

```diff
@@ -238,12 +238,65 @@ function genericPrintNoParens(path, options, print) {
     );
   case "BinaryExpression":
   case "LogicalExpression":
+    var parent = path.getParentNode();
+    var isStartOfLogicalExpressionChain = parent ? parent.type !== 'LogicalExpression' : false;
```

Comment:
We have to know when we're entering a LogicalExpression change vs recursing in the middle of a chain. This was my first idea, but it fails when you have a LogicalExpression in the middle of a LogicalExpression.

Ex:
```js
parent.type === "MemberExpression"
&& isNumber.check(node.value)
&& name === "object"
&& (
  nestedOne || nestedTwo || nestedThree || nestedFour || nestedFive || nestedSix || nestedSeven || nestedEight
);
```

I'd expect this to be formatted as:

```js
(
  parent.type === "MemberExpression" &&
  isNumber.check(node.value) &&
  name === "object" &&
  (
    nestedOne || 
    nestedTwo || 
    nestedThree || 
    nestedFour || 
    nestedFive || 
    nestedSix || 
    nestedSeven || 
    nestedEight
  )
)
```

I'm unsure how to detect the second group...